### DISCUSSION TITLE
Can we improve this generic errror `got 1 error(s) in grl the script`

### DISCUSSION THREAD

**Discussion by jinagamvasubabu** [hyperjumptech/grule-rule-engine#246] [Q&A] [answered]
Hi @newm4n , @niallnsec 

Context:
I have around 1000 rules among which one of the rule is having an error and I get this error while building the resource from my existing rules.

```go
err := ruleBuilder.BuildRuleFromResource(kbType, constants.KnowledgeBaseVersion, byteArr)
```

Error is:
```go
got 1 error(s) in grl the script
```

It would be tough for me to go through all rules and figure out which rule is having the above issue and also the error is so generic and doesn't point to the rule which is having the issue. Can we improve this error message ? if yes, can you give me some pointers, so i can work on it.

**Comment by niallnsec** [comment, marked as answer]
The returned error should be for type *GruleErrorReporter, so you can type assert and access the embedded Errors array which will hopefully give you a more specific and useful error.

**Reply by jinagamvasubabu** [reply]
Hi @niallnsec, Thanks for your quick reply, I have tried this way before and what i got is 
```go
grl error on 3321:200 mismatched input ')' expecting {'-', '(', TRUE, FALSE, NIL_LITERAL, '!', SIMPLENAME, DQUOTA_STRING, SQUOTA_STRING, DECIMAL_FLOAT_LIT, HEX_FLOAT_LIT, DEC_LIT, HEX_LIT, OCT_LIT}
```
I load rules from DB and this 3321:200 points to GRL file that creates using these rules coming from DB, so these line numbers doesn't help me to identify the rule. Is there any way to show any other identifier like rule name or rule id with this error message, is it possible ?

One way to fix is, validate the rule dsl before even entering into system :)
